app/uk/gov/hmrc/agentepayeregistration/repository/AgentEpayeRegistrationRepository.scala

```diff
@@ -37,23 +38,43 @@ class AgentEpayeRegistrationRepository @Inject()(mongo: ReactiveMongoComponent)
     RegistrationDetails.registrationDetailsFormat,
     ReactiveMongoFormats.objectIdFormats) {
 
+
   override def indexes: Seq[Index] =
-    Seq(Index(key = Seq("agentReference" -> IndexType.Ascending), name = Some("agentRefIndex"), unique = true))
+    Seq(Index(key = Seq("agentReference" -> IndexType.Ascending), name = Some("agentRefIndex"), unique = true),
+      Index(key = Seq("createdDateTime" -> IndexType.Ascending), name = Some("createdDateTimeIndex"), unique = false))
 
-  val initialAgentReference = "HX2000"
+  val initialAgentReference: String = "HX2000"
 
-  def create(request: RegistrationRequest)(implicit ec: ExecutionContext): Future[AgentReference] = {
-    val mongoCodeDuplicateKey = 11000
+  def create(request: RegistrationRequest, createdDate: DateTime = DateTime.now(DateTimeZone.UTC))
+            (implicit ec: ExecutionContext): Future[AgentReference] = {
+    val mongoCodeDuplicateKey: Int = 11000
 
     for {
       maybeRegDetails <- collection.find(obj()).sort(obj("agentReference" -> -1)).one[RegistrationDetails]
       nextAgentRef = maybeRegDetails match {
         case Some(regDetails) => regDetails.agentReference.newReference
         case None => AgentReference(initialAgentReference)
       }
-      _ <- insert(RegistrationDetails(nextAgentRef, request)) recover {
+      _ <- insert(RegistrationDetails(nextAgentRef, request, createdDate)) recover {
         case error: DatabaseException if error.code.contains(mongoCodeDuplicateKey) => create(request)
       }
     } yield nextAgentRef
   }
+
+  def findRegistrations(dateTimeFrom: DateTime, dateTimeTo: DateTime)
+                       (implicit ec: ExecutionContext): Future[List[RegistrationDetails]] = {
+    require(!dateTimeTo.isBefore(dateTimeFrom), "to date is before from date")
+
+    val queryFilter: play.api.libs.json.JsObject = obj(
+      "createdDateTime" -> obj(
+        "$gte" -> obj("$date" -> dateTimeFrom.getMillis),
+        "$lte" -> obj("$date" -> dateTimeTo.getMillis)
+      )
+    )
+
+    collection.find(queryFilter)
+      .sort(obj("createdDateTime" -> 1))
+      .cursor[RegistrationDetails]()
+      .collect[List](Integer.MAX_VALUE)
```

Comment:
You could try
```scala
 find("createdDateTime" -> obj(
      "$gte" -> obj("$date" -> dateTimeFrom.getMillis),
      "$lte" -> obj("$date" -> dateTimeTo.getMillis)
    )).map(_.sortBy(_.createdDateTime)(Ordering.fromLessThan(_ isBefore _)))
```

The find method is from `uk.gov.hmrc.mongo.ReactiveRepository` class

Reply:
that method is expecting a JsValueWrapper)* which is quite lengthy to change.